networking/metallb/metallb-configure-services.adoc

```diff
@@ -0,0 +1,162 @@
+  Warning  AllocationFailed  3m16s  metallb-controller  Failed to allocate IP for "default/invalid-request": "4.3.2.1" is not allowed in config
+----
+
+[id="request-ip-address-from-pool_{context}"]
+== Request an IP address from a specific pool
+
+To assign an IP address from a specific range, but you are not concerned with the specific IP address, then you can use the `metallb.universe.tf/pool-name` annotation to request an IP address from the specified address pool.
+
+.Example service YAML for an IP address from a specific pool
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: <service_name>
+  annotations:
+    metallb.universe.tf/pool-name: <address_pool_name>
+spec:
+  selector:
+    <label_key>: <label_value>
+  ports:
+    - port: 8080
+      targetPort: 8080
+      protocol: TCP
+  type: LoadBalancer
+----
+
+If the address pool that you specify for `<address_pool_name>` does not exist, MetalLB attempts to assign an IP address from any pool that permits automatic assignment.
+
+[id="accept-any-ip-address_{context}"]
+== Accept any IP address
+
+By default, address pools are configured to permit automatic assignment.
+MetalLB assigns an an IP address from these address pools.
+
+To accept any IP address from any pool that is configured for automatic assignment, no special annotation or configuration is required.
+
+.Example service YAML for accepting any IP address
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: <service_name>
+spec:
+  selector:
+    <label_key>: <label_value>
+  ports:
+    - port: 8080
+      targetPort: 8080
+      protocol: TCP
+  type: LoadBalancer
+----
+
+[id="share-specific-ip-address_{context}"]
+== Share a specific IP address
+
+By default, services do not share IP addresses.
+However, if you need to colocate services on a single IP address, you can enable selective IP sharing by adding the `metallb.universe.tf/allow-shared-ip` annotation to the services.
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: service-http
+  annotations:
+    metallb.universe.tf/address-pool: doc-example
+    metallb.universe.tf/allow-shared-ip: "web-server-svc"  <1>
+spec:
+  ports:
+    - name: http
+      port: 80  <2>
+      protocol: TCP
+      targetPort: 8080
+  selector:
+    <label_key>: <label_value>  <3>
+  type: LoadBalancer
+  loadBalancerIP: 172.31.249.7  <4>
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: service-https
+  annotations:
+    metallb.universe.tf/address-pool: doc-example
+    metallb.universe.tf/allow-shared-ip: "web-server-svc"  <1>
+spec:
+  ports:
+    - name: https
+      port: 443  <2>
+      protocol: TCP
+      targetPort: 8080
+  selector:
+    <label_key>: <label_value>  <3>
+  type: LoadBalancer
+  loadBalancerIP: 172.31.249.7  <4>
+----
+<1> Specify the same value for the `metallb.universe.tf/allow-shared-ip` annotation. This value is referred to as the _sharing key_.
```

Comment:
Did you want to make these auto numbering as well like another file above? Just for consistency. It's not necessary :-)

Reply:
I suppose I could, but I repeated the numbering as a reminder to myself that there's something weird--the code block has callouts repeat.